shm/player.cpp

```diff
@@ -0,0 +1,27 @@
+#include "player.hpp"
+
+Player::Player(std::unique_ptr<Ship> ship, size_t money, size_t availableSpace)
+    : ship_(std::move(ship)), money_(money), availableSpace_(availableSpace) {}
+
+Cargo* Player::getCargo(size_t index) const {
+    if (index < (ship_->getCargo().size())) {
+        return &ship_->getCargo()[index];
+    } else {
+        return nullptr;
+    }
+}
+
+void Player::countAvailableSpace() {
+    const std::vector<Cargo> temp = ship_->getCargo();
```

Comment:
I think, you don't have to create variable temp and work on created copy.

Reply:
+1